src/server/server.rs

```diff
@@ -48,12 +50,15 @@ pub struct Server<T: RaftStoreRouter + 'static, S: StoreAddrResolver + 'static>
     // For sending/receiving snapshots.
     snap_mgr: SnapManager,
     snap_worker: Worker<SnapTask>,
+    metrics_flusher: Option<MetricsFlusher>,
 }
 
 impl<T: RaftStoreRouter, S: StoreAddrResolver + 'static> Server<T, S> {
+    #[allow(too_many_arguments)]
     pub fn new(cfg: &Config,
                region_split_size: usize,
                storage: Storage,
+               engine: Option<Arc<DB>>,
```

Comment:
my fault, it is ugly to start rocksdb flusher in Server. maybe it is better to start the flusher directly in the tikv-server.rs after the storage starts.